fuzz.go

```diff
@@ -120,14 +196,46 @@ func Fuzz(data []byte) int {
 			}
 			return z.Lsh(x, n)
 		}
-		checkOp(lsh, bigLsh, x, y)
+		checkDualArgOp(lsh, bigLsh, x, y)
 
 	case opAdd:
-		checkOp((*Int).Add, (*big.Int).Add, x, y)
+		checkDualArgOp((*Int).Add, (*big.Int).Add, x, y)
 
 	case opSub:
-		checkOp((*Int).Sub, (*big.Int).Sub, x, y)
+		checkDualArgOp((*Int).Sub, (*big.Int).Sub, x, y)
+	default:
+		return 0
+	}
+	return 1
+}
+
+func bigMulMod(b1, b2, b3, b4 *big.Int) *big.Int {
+	return b1.Mod(big.NewInt(0).Mul(b2, b3), b4)
+}
+
+func intMulMod(f1, f2, f3, f4 *Int) *Int {
+	return f1.MulMod(f2, f3, f4)
+}
+
+func Fuzz3Args(data []byte) int {
+	if len(data) != 1+32+32+23 {
+		return 0
 	}
+	op := data[0]
 
-	return 0
+	var x, y, z Int
+	x.SetBytes(data[1:33])
+	y.SetBytes(data[33:65])
+	z.SetBytes(data[65:])
+
+	switch op {
```

Comment:
I was actually thinking that maybe instead of switching on the first byte just run all operators in a row.

Reply:
Yes, good idea